modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java

```diff
@@ -931,6 +940,69 @@ public void deleteMessageMetadataFromQueue(final String storageQueueName,
     /**
      * {@inheritDoc}
      */
+    public void deleteMessageMetadataAndContent(final String storageQueueName,
+                                                List<AndesRemovableMetadata> messagesToRemove)
+            throws AndesException {
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        PreparedStatement contentRemovalPreparedStatment = null;
+
+        Context context = DataAccessMatrixManager.addAndGetTimer(MatrixConstants.DELETE_MESSAGE_METADATA_AND_CONTENT, this).
+                start();
+
+        try {
+            int queueID = getCachedQueueID(storageQueueName);
+
+            connection = getConnection();
+            connection.setAutoCommit(false);
+            preparedStatement = connection
+                    .prepareStatement(RDBMSConstants.PS_DELETE_METADATA_FROM_QUEUE);
+            contentRemovalPreparedStatment = connection.prepareStatement(RDBMSConstants.PS_DELETE_MESSAGE_PARTS);
+            for (AndesRemovableMetadata md : messagesToRemove) {
+                //add parameter to delete metadata
+                preparedStatement.setInt(1, queueID);
+                preparedStatement.setLong(2, md.getMessageID());
+                preparedStatement.addBatch();
+
+                //add parameters to delete content
+                contentRemovalPreparedStatment.setLong(1, md.getMessageID());
+                contentRemovalPreparedStatment.addBatch();
+            }
+            preparedStatement.executeBatch();
+            contentRemovalPreparedStatment.executeBatch();
+            connection.commit();
+
+            if (log.isDebugEnabled()) {
```

Comment:
Did not use cascade deletes because there are instances where we only delete metadata
For instance, when moving messages to dead letter channel